[CI] StatelessIT testBackgroundMergeCommitAfterRelocationHasStartedDoesNotSendANewCommitNotification failing

**Build Scans:**
- [elasticsearch-intake #42424 / part2](https://gradle-enterprise.elastic.co/s/ipvmg6adtt6ie)
- [elasticsearch-pull-request #158760 / part-2](https://gradle-enterprise.elastic.co/s/nawucmhlbbk3u)
- [elasticsearch-pull-request #158748 / part-2](https://gradle-enterprise.elastic.co/s/jgob7zw76rcd6)
- [elasticsearch-pull-request #158750 / part-2](https://gradle-enterprise.elastic.co/s/usonbs3fwk5tm)
- [elasticsearch-pull-request #158748 / part-2](https://gradle-enterprise.elastic.co/s/ee37ohn6gwdbm)
- [elasticsearch-pull-request #158750 / part-2](https://gradle-enterprise.elastic.co/s/jywh53xpehkqc)
- [elasticsearch-pull-request #158738 / part-2](https://gradle-enterprise.elastic.co/s/lxhalwwffg4yk)
- [elasticsearch-pull-request #158739 / part-2](https://gradle-enterprise.elastic.co/s/l4xtwwokhs4gq)
- [elasticsearch-pull-request #158737 / part-2](https://gradle-enterprise.elastic.co/s/fdaynqlywwo2q)
- [elasticsearch-pull-request #158736 / part-2](https://gradle-enterprise.elastic.co/s/wgljxke5qk2zm)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:stateless:internalClusterTest" --tests "org.elasticsearch.xpack.stateless.StatelessIT.testBackgroundMergeCommitAfterRelocationHasStartedDoesNotSendANewCommitNotification" -Dtests.seed=D07364397C3DB284 -Dtests.locale=ff-Adlm-GW -Dtests.timezone=Asia/Almaty -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.stateless.StatelessIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testBackgroundMergeCommitAfterRelocationHasStartedDoesNotSendANewCommitNotification'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: unexpected
```

**Issue Reasons:**
- [main] 2 consecutive failures in test testBackgroundMergeCommitAfterRelocationHasStartedDoesNotSendANewCommitNotification
- [main] 1 consecutive failures in step part2
- [main] 1 consecutive failures in step part-2
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 1 consecutive failures in pipeline elasticsearch-pull-request
- [main] 25 failures in test testBackgroundMergeCommitAfterRelocationHasStartedDoesNotSendANewCommitNotification (2.8% fail rate in 901 executions)
- [main] 3 failures in step part2 (2.2% fail rate in 137 executions)
- [main] 22 failures in step part-2 (3.7% fail rate in 587 executions)
- [main] 3 failures in pipeline elasticsearch-intake (2.2% fail rate in 137 executions)
- [main] 11 failures in pipeline elasticsearch-pull-request (1.9% fail rate in 577 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -773,9 +773,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.expression.function.aggregate.SumSerializationTests
   method: testSerializeSumWithOverflowingLongSupplier
   issue: https://github.com/elastic/elasticsearch/issues/152506
-- class: org.elasticsearch.xpack.stateless.StatelessIT
-  method: testBackgroundMergeCommitAfterRelocationHasStartedDoesNotSendANewCommitNotification
-  issue: https://github.com/elastic/elasticsearch/issues/152516
 - class: org.elasticsearch.transport.RemoteClusterClientTests
   method: testConnectAndExecuteRequest
   issue: https://github.com/elastic/elasticsearch/issues/152518

--- a/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/StatelessIT.java
+++ b/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/StatelessIT.java
@@ -964,7 +964,9 @@ public class StatelessIT extends AbstractStatelessPluginIntegTestCase {
                 // Force merge checks if the engine is still open at the end, and sometimes it might
                 // throw an AlreadyClosedException even after the commit is already processed by ShardCommitState
             } catch (IOException e) {
-                fail(e);
+                if (e.getCause() instanceof MergePolicy.MergeAbortedException == false) {
+                    fail(e);
+                }
             }
         }, "force-merge-thread");
         forceMergeThread.start();

